vsphere/resource_vsphere_host_virtual_switch_test.go

```diff
@@ -0,0 +1,254 @@
+	if os.Getenv("VSPHERE_HOST_NIC0") == "" {
+		t.Skip("set VSPHERE_HOST_NIC0 to run vsphere_host_virtual_switch acceptance tests")
+	}
+	if os.Getenv("VSPHERE_HOST_NIC1") == "" {
+		t.Skip("set VSPHERE_HOST_NIC1 to run vsphere_host_virtual_switch acceptance tests")
+	}
+	if os.Getenv("VSPHERE_ESXI_HOST") == "" {
+		t.Skip("set VSPHERE_ESXI_HOST to run vsphere_host_virtual_switch acceptance tests")
+	}
+}
+
+func testAccResourceVSphereHostVirtualSwitchExists(expected bool) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources["vsphere_host_virtual_switch.switch"]
+		if !ok {
+			return errors.New("vsphere_host_virtual_switch.switch not found in state")
+		}
+
+		client := testAccProvider.Meta().(*govmomi.Client)
+		id := rs.Primary.ID
+		host := os.Getenv("VSPHERE_ESXI_HOST")
+		datacenter := os.Getenv("VSPHERE_DATACENTER")
+		timeout := time.Minute * 5
+		_, err := hostVSwitchFromName(client, id, host, datacenter, timeout)
+		if err != nil {
+			if err.Error() == fmt.Sprintf("vSwitch %s not found on host %s", id, host) && !expected {
+				// Expected missing
+				return nil
+			}
+			return err
+		}
+		if !expected {
+			return fmt.Errorf("expected vSwitch %s to still be missing", id)
+		}
+		return nil
+	}
+}
+
+func testAccResourceVSphereHostVirtualSwitchConfig() string {
+	return fmt.Sprintf(`
+variable "esxi_host" {
+  type    = "string"
+  default = "%s"
+}
+
+variable "datacenter" {
+  type    = "string"
+  default = "%s"
+}
+
+variable "host_nic0" {
+  type    = "string"
+  default = "%s"
+}
+
+variable "host_nic1" {
+  type    = "string"
+  default = "%s"
+}
+
+resource "vsphere_host_virtual_switch" "switch" {
+  name             = "vSwitchTerraformTest"
+  host             = "${var.esxi_host}"
+  datacenter       = "${var.datacenter}"
+  network_adapters = ["${var.host_nic0}", "${var.host_nic1}"]
+
+  active_nics  = ["${var.host_nic0}", "${var.host_nic1}"]
+  standby_nics = []
+}
+`, os.Getenv("VSPHERE_ESXI_HOST"), os.Getenv("VSPHERE_DATACENTER"), os.Getenv("VSPHERE_HOST_NIC0"), os.Getenv("VSPHERE_HOST_NIC1"))
+}
+
+func testAccResourceVSphereHostVirtualSwitchConfigSingleNIC() string {
+	return fmt.Sprintf(`
+variable "esxi_host" {
+  type    = "string"
+  default = "%s"
+}
+
+variable "datacenter" {
+  type    = "string"
+  default = "%s"
+}
+
+variable "host_nic0" {
+  type    = "string"
+  default = "%s"
+}
+
+resource "vsphere_host_virtual_switch" "switch" {
+  name             = "vSwitchTerraformTest"
+  host             = "${var.esxi_host}"
+  datacenter       = "${var.datacenter}"
+  network_adapters = ["${var.host_nic0}"]
+
+  active_nics  = ["${var.host_nic0}"]
+  standby_nics = []
+}
+`, os.Getenv("VSPHERE_ESXI_HOST"), os.Getenv("VSPHERE_DATACENTER"), os.Getenv("VSPHERE_HOST_NIC0"))
+}
+
+func testAccResourceVSphereHostVirtualSwitchConfigStandbyLink() string {
+	return fmt.Sprintf(`
+variable "esxi_host" {
+  type    = "string"
```

Comment:
_minor_ we can remove the types here

Reply:
Thoughts [above](https://github.com/terraform-providers/terraform-provider-vsphere/pull/138#discussion_r135161820) on this one.